packages/cloudflare-optimizer-scripts/src/index.js

```diff
@@ -66,14 +66,23 @@ async function handleRequest(event, config) {
 
   let request = event.request;
   const url = new URL(request.url);
+
   if (isReverseProxy(config)) {
     url.hostname = config.proxy.origin;
   }
+
   request = new Request(url.toString(), request);
 
   // Immediately return if not GET.
   if (request.method !== 'GET') {
-    return fetch(request);
+    const response = await fetch(request, {redirect: 'manual'});
+
+    // For reverse proxy based responses, Location header needs to be rewritten in some cases
+    if (isReverseProxy(config) && isRedirect(response)) {
```

Comment:
feel free to ignore, but technically the code would be faster if we separated the two checks. because the `fetch` doesn't ever need to be awaited here in the non-proxy case.

Something like:
```js
if (!isReverseProxy(config)) {
  return fetch(request);
}
const response = await fetch(request, {redirect: 'manual'});
return isRedirect(response) 
    ? rewriteRedirectHeader(response, config)
    : response
}
```

Reply:
Thank you so much for all your guidance, This is a genius way to structure logic in order to optimize. 
I have made the change